protocol/builder/api_versions_response_builder.go

```diff
@@ -0,0 +1,56 @@
+package builder
+
+import (
+	kafkaapi "github.com/codecrafters-io/kafka-tester/protocol/api"
+)
+
+type ApiVersionsResponseBuilder struct {
+	version        int16
+	errorCode      int16
+	apiKeys        []kafkaapi.ApiKeyVersionSupport
+	throttleTimeMs int32
+}
+
+func NewApiVersionsResponseBuilder() *ApiVersionsResponseBuilder {
+	return &ApiVersionsResponseBuilder{
+		version:        3,
+		errorCode:      0,
+		throttleTimeMs: 0,
+		apiKeys:        []kafkaapi.ApiKeyVersionSupport{},
+	}
+}
+
+func (b *ApiVersionsResponseBuilder) WithErrorCode(errorCode int16) *ApiVersionsResponseBuilder {
+	b.errorCode = errorCode
+	return b
+}
+
+func (b *ApiVersionsResponseBuilder) WithThrottleTimeMs(throttleTimeMs int32) *ApiVersionsResponseBuilder {
+	b.throttleTimeMs = throttleTimeMs
+	return b
+}
+
+func (b *ApiVersionsResponseBuilder) AddApiKeyVersionSupport(apiKey int16, minVersion int16, maxVersion int16) *ApiVersionsResponseBuilder {
```

Comment:
```suggestion
func (b *ApiVersionsResponseBuilder) AddApiKeyEntry(apiKey int16, minVersion int16, maxVersion int16) *ApiVersionsResponseBuilder {
```

(or `AddApiKeyListItem`, `AddApiKey` ...)